languages/rust/generic/Parse_rust_tree_sitter.ml

```diff
@@ -82,11 +82,46 @@ type rust_macro_item =
   | MacTree of rust_macro_item list G.bracket
   | MacTreeBis of rust_macro_item list G.bracket * G.ident option * G.tok
 
-(* Note that the commas are considered like any other tokens in a Rust macro;
- * they are not separators between rust_macro_items.
- *)
 let rec macro_items_to_anys (xs : rust_macro_item list) : G.any list =
-  xs |> Common.map macro_item_to_any
+  (* Note that the commas are considered like any other tokens in a Rust macro;
+     * they are not separators between rust_macro_items.
+  *)
+  (* Thus, we need this machinery in order to parse a list of macro items
+     to see if the macro invocation looks like a normal expression. For instance,
+     if a sequence of macro items looks like
```

Comment:
Is there a risk to get this wrong?

Reply:
the default behavior is we fail to parse as a normal sequence of arguments, and the translation proceeds as before. if we get it wrong, then we might turn it into `Args` when we shouldn't, and therefore pass normal arguments into the macro. However, otherwise it would just be a sequence of `any`s, like `G.E, G.Tk, G.E, G.Tk` which I doubt would have any particular semantic behavior. so i don't think there's any risk.